src/components/playback/playbackmanager.js

```diff
@@ -3304,10 +3304,10 @@ class PlaybackManager {
                     const startTime = getCurrentTicks(player) || streamInfo.playerStartPositionTicks;
 
                     changeStream(player, startTime, {
-                        // force transcoding
+                        // force transcoding and only allow remuxing for remote source like liveTV
                         EnableDirectPlay: false,
                         EnableDirectStream: false,
-                        AllowVideoStreamCopy: false,
+                        AllowVideoStreamCopy: streamInfo.item.LocationType === 'Remote',
```

Comment:
What happens when it fails with `AllowAudioStreamCopy: false`?
Will it try `AllowVideoStreamCopy: false`?

Like:
1. Direct play.
2. Allow copy video, allow copy audio. 👈 LiveTV retry
3. Allow copy video, deny copy audio.
4. Deny copy video, deny copy audio.

Or will it skip 4 from the above?

Reply:
It does not need to do anymore retry. This will be the last one because server-side profile will be limited to ts+h264, and the server will do transcode if the original source is not compatible with this profile. AudioStream is not relevant and the server can transcode it at will.

If there is a client that does not even play h264+ts then it basically cannot play anything.

Reply:
With prefer fMP4, there will also be `h264-mp4`.

AFAIK, the last retry is for when we can't reliably generate a device profile. There may be some quirks. 🤷

Reply:
> With prefer fMP4, there will also be h264-mp4.

No, I intentionally override it on server. The client preference does nothing. The global setting should not affect LiveTV at all due to complicated compatibility reasons. 

> AFAIK, the last retry is for when we can't reliably generate a device profile. 

For LiveTV this is almost always the case, because the fallback is super common as the direct play of remote url almost always fail for hls.js, and Safari can play it a little bit, but still not all, which the fallback to server will be very, very common. 

> There may be some quirks. 🤷

LiveTV itself is a quirk.

Reply:
> No, I intentionally override it on server. The client preference does nothing. The global setting should not affect LiveTV at all due to complicated compatibility reasons.

And then they provide you with LiveTV with HEVC. 🙃

> For LiveTV this is almost always the case, because the fallback is super common as the direct play of remote url almost always fail for hls.js, and Safari can play it a little bit, but still not all, which the fallback to server will be very, very common.

I mean, it's gonna be like:
1. Direct play.
2. Direct play, remux or transcode (only if codec not supported).
3. Direct stream or transcode (only if codec not supported).

But no full (forced) transcoding as a last chance. In case of some unexpected media error/player quirk/solar storm.

When I tested a direct HLS profile for LiveTV in Chrome, it played directly (no retires) or didn't play due to inaccessibility. Probably because the server just returned the URL without any validation.

Reply:
> And then they provide you with LiveTV with HEVC. 🙃

This will trigger transcode, what's the problem? The server is not as dumb as you think. You enable copy does not mean server will always copy. It will check if it really meet the request. 

Enabling HEVC LiveTV is way beyond the 10.9 goal. We cannot do it right without change of the API.

> But no full (forced) transcoding as a last chance. In case of some unexpected media error/player quirk/solar storm.

The users are complaining because we are doing forced transcode and skipping remux. Do you have a better way that allows testing for remux for LiveTV that actually works? Yes forced transcoding is safe, but user hates transcode very very very much.

> When I tested a direct HLS profile for LiveTV in Chrome, it played directly (no retires) or didn't play due to inaccessibility. Probably because the server just returned the URL without any validation.

Please, it did not play for **a lot** of streams. The fallback is **very very common**. Please don't pretend all livetv streamers provide good hls streams that can be played by browser that easy. Just look at the server repo and search for live tv issue and you will know that. Please don't make naive assumption about how good the direct play will work.

Reply:
> This will trigger transcode, what's the problem? The server is not as dumb as you think. You enable copy does not mean server will always copy. It will check if it really meet the request.

I know how StreamBuilder works and what is a mess. 🙃
I meant, users will complain "why HEVC -> H264 ?" and you will have to remove that "not use fMP4" block.

> The users are complaining because we are doing forced transcode and skipping remux. Do you have a better way that allows testing for remux for LiveTV that actually works? Yes forced transcoding is safe, but user hates transcode very very very much.

Just add this 4th retry in addition to the one you added in this PR so it will play with full transcode instead of playback failure.

> Please, it did not play for **a lot** of streams. The fallback is **very very common**. Please don't pretend all livetv streamers provide good hls streams that can be played by browser that easy. Just look at the server repo and search for live tv issue and you will know that. Please don't make naive assumption about how good the direct play will work.

I don't make any assumptions, I am saying how it worked when I added direct HLS profile.
AFAIK, the server does probing of LiveTV and stores media info in JSON-file somewhere (_I don't know how to refresh it, btw_).
StreamBuilder was getting actual media info to make right decision. The problem here is that the probing may fail - in my case, due to unavailability.
In 10.8, I once had to reorder this probing, placing it before the stream building. In unstable, it worked without reordering.

Reply:
> I meant, users will complain "why HEVC -> H264 ?" and you will have to remove that "not use fMP4" block.

I won't. I will tell them it is not supported. Wait for 10.10. You misunderstood the whole point here because it seems like you are totally unfamiliar with what the users are facing:

```
// Remove all fmp4 transcoding profiles, because it causes playback error and/or A/V sync issues
// Notably: Some channels won't play on FireFox and LG webOS even after transcoding
// Some channels from HDHomerun will experience A/V sync issues due to remux into fmp4
```

Plus, HEVC LiveTV never really worked before, even in 10.8 due to fmp4 support is really flawed at that time. I'm solving "but it works in 10.8" issue, not adding fancy new functions. For ATSC 3.0 TV stream the server has to support basic AC-4 audio which is also missing. So HEVC stream can definitely wait. 

> StreamBuilder was getting actual media info to make right decision.

If you are getting here the direct play attempt is already failed.

The most cursed part of this is, the actual media info can also be wrong. The bitrate could be null and causing the client to request extremely high bitrate, and if the transcode is triggered it will try to encode at 1000mbps. That's why LiveTV needs a separate encoding upper limit to overcome this situation. When the source bitrate is unavailable we need a sane value that will not fail encoders and players. I currently set the hard limit at 30mbps.

Reply:
> I won't. I will tell them it is not supported. Wait for 10.10.

Where did I say 10.9?

> If you are getting here the direct play attempt is already failed.

Ok. So it was working as it should, just for me.

> The most cursed part of this is, the actual media info can also be wrong. The bitrate could be null and causing the client to request extremely high bitrate, and if the transcode is triggered it will try to encode at 1000mbps. That's why LiveTV needs a separate encoding upper limit to overcome this situation. When the source bitrate is unavailable we need a sane value that will not fail encoders and players. I currently set the hard limit at 30mbps.

And retries are for these cases (abnormal media and hacks).
I am not against this PR, I am against removing forced transcoding (the last retry).

Reply:
> Ok. So it was working as it should, just for me.

This is normal for LiveTV. It is impossible to test what the providers are doing because most of them will do it wrong. The "supposed to work" way almost always don't work.

> I am not against this PR, I am against removing forced transcoding (the last retry).

Added.